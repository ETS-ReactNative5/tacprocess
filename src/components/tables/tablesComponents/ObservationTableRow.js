import React from 'react'
import propTypes from 'prop-types'

const ObservationTabelRow = ({observation}) => {
  if (observation == null || Object.keys(observation).length === 0) return null
  return (
    <tr key={ observation.priority }>
      <td>P{ observation.priority }</td>
      <td>{ observation.percentage }%</td>
    </tr>
  )
}

ObservationTabelRow.propTypes = {
  observation: propTypes.object.isRequired
}
export default ObservationTabelRow
