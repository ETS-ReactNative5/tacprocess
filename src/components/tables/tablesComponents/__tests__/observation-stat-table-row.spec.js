import React from "react"
import { mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import ObservationTableRow from "../ObservationTableRow"

const proposals = [
  { proposalCode: 'Code-1',
    title: 'Title-1',
    principalInvestigator: { givenName: 'PI-1', familyName: 'PI-1'},
    liaisonAstronomer: { givenName: 'LA-1', familyName: 'LA-1'},
    status: 'Status-1',
    statusComment: 'Comment-1',
    timeAllocations: [{ priority: 1, amount: 50000, semester: '2018-1', partnerCode: 'RSA' }],
    observations: [{ status: 'ACCEPTED', block: { length: 2000, priority: 1, semester: '2018-1'} }]
  }
]

// Checking if the ObservationTableRow Component renders correctly for different input
describe("ObservationTableRow Component", () => {
  // Enzyme method testing if it renders correctly with empty values of the proposals
  it("Render the ObservationTableRow Component having unpopulated props with no errors", () => {
    const rendered = mount(<ObservationTableRow
      proposal= {{}}
      semester={''}
    />)
    expect(shallowToJson(rendered)).toMatchSnapshot()
  })
  
  it("Render the ObservationTableRow Component having populated props with no errors", () => {
    const rendered = mount(<ObservationTableRow
      proposal= {proposals[0]}
      semester={'2018-1'}
    />)
    expect(shallowToJson(rendered)).toMatchSnapshot()
  })
})
