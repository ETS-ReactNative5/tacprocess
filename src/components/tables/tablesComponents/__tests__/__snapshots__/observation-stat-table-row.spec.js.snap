// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationTableRow Component Render the ObservationTableRow Component having populated props with no errors 1`] = `
<ObservationTabelRow
  proposal={
    Object {
      "liaisonAstronomer": Object {
        "familyName": "LA-1",
        "givenName": "LA-1",
      },
      "observations": Array [
        Object {
          "block": Object {
            "length": 2000,
            "priority": 1,
            "semester": "2018-1",
          },
          "status": "ACCEPTED",
        },
      ],
      "principalInvestigator": Object {
        "familyName": "PI-1",
        "givenName": "PI-1",
      },
      "proposalCode": "Code-1",
      "status": "Status-1",
      "statusComment": "Comment-1",
      "timeAllocations": Array [
        Object {
          "amount": 50000,
          "partnerCode": "RSA",
          "priority": 1,
          "semester": "2018-1",
        },
      ],
      "title": "Title-1",
    }
  }
  semester="2018-1"
/>
`;

exports[`ObservationTableRow Component Render the ObservationTableRow Component having unpopulated props with no errors 1`] = `
<ObservationTabelRow
  proposal={Object {}}
  semester=""
/>
`;
